### PR TITLE
Add support for Sve.DuplicateSelectedScalarToVector()

### PR DESCRIPTION
**Pull request by SwapnilGaikwad**
Contribute towards https://github.com/dotnet/runtime/issues/99957.

### CODE REVIEW

**Comment by SwapnilGaikwad** [comment]
@a74nh @kunalspathak @dotnet/arm64-contrib @arch-arm64-sve

**Comment by SwapnilGaikwad** [comment]
There are a few assertion failures, related to LDR/STR. Taking a further look.

<details>
<summary> Stress test results </summary>

```
===================Running default===================
------------------- {} -------------------
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.SveDuplicateSelectedScalarToVector_float() : 17
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.SveDuplicateSelectedScalarToVector_double() : 17
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.SveDuplicateSelectedScalarToVector_sbyte() : 17
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.SveDuplicateSelectedScalarToVector_short() : 17
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.SveDuplicateSelectedScalarToVector_int() : 17
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.SveDuplicateSelectedScalarToVector_long() : 17
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.SveDuplicateSelectedScalarToVector_byte() : 17
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.SveDuplicateSelectedScalarToVector_ushort() : 17
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.SveDuplicateSelectedScalarToVector_uint() : 17
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.SveDuplicateSelectedScalarToVector_ulong() : 17
===================Running jitstress===================
------------------- {'JitMinOpts': '1'} -------------------
------------------- {'JitStress': '1'} -------------------
------------------- {'JitStress': '2'} -------------------
------------------- {'JitStress': '1', 'TieredCompilation': '1'} -------------------
------------------- {'JitStress': '2', 'TieredCompilation': '1'} -------------------
------------------- {'TailcallStress': '1'} -------------------
------------------- {'ReadyToRun': '0'} -------------------
===================Running jitstressregs===================
------------------- {'JitStressRegs': '1'} -------------------
------------------- {'JitStressRegs': '2'} -------------------
------------------- {'JitStressRegs': '3'} -------------------
------------------- {'JitStressRegs': '4'} -------------------
------------------- {'JitStressRegs': '8'} -------------------
------------------- {'JitStressRegs': '0x10'} -------------------
------------------- {'JitStressRegs': '0x80'} -------------------
------------------- {'JitStressRegs': '0x1000'} -------------------
------------------- {'JitStressRegs': '0x2000'} -------------------
===================Running jitstress2-jitstressregs===================
------------------- {'JitStress': '2', 'JitStressRegs': '1'} -------------------
------------------- {'JitStress': '2', 'JitStressRegs': '2'} -------------------
Errors:

Assert failure(PID 2894704 [0x002c2b70], Thread: 2894704 [0x2c2b70]): Assertion failed 'isValidSimm<9>(emitGetInsSC(id))' in 'JIT.HardwareIntrinsics.Arm._Sve.Program:SveDuplicateSelectedScalarToVector_double()' during 'Generate code' (IL size 115; hash 0x749c6f16; FullOpts)

    File: /home/user/dotnet/runtime/src/coreclr/jit/emitarm64sve.cpp:14192
    Image: /home/user/dotnet/runtime/artifacts/tests/coreclr/linux.arm64.Checked/Tests/Core_Root/corerun


------------------- {'JitStress': '2', 'JitStressRegs': '3'} -------------------
------------------- {'JitStress': '2', 'JitStressRegs': '4'} -------------------
------------------- {'JitStress': '2', 'JitStressRegs': '8'} -------------------
------------------- {'JitStress': '2', 'JitStressRegs': '0x10'} -------------------
------------------- {'JitStress': '2', 'JitStressRegs': '0x80'} -------------------
Errors:

Assert failure(PID 2894838 [0x002c2bf6], Thread: 2894838 [0x2c2bf6]): Assertion failed 'isValidSimm<9>(emitGetInsSC(id))' in 'JIT.HardwareIntrinsics.Arm._Sve.Program:SveDuplicateSelectedScalarToVector_double()' during 'Generate code' (IL size 115; hash 0x749c6f16; FullOpts)

    File: /home/user/dotnet/runtime/src/coreclr/jit/emitarm64sve.cpp:14192
    Image: /home/user/dotnet/runtime/artifacts/tests/coreclr/linux.arm64.Checked/Tests/Core_Root/corerun


------------------- {'JitStress': '2', 'JitStressRegs': '0x1000'} -------------------
------------------- {'JitStress': '2', 'JitStressRegs': '0x2000'} -------------------
```
</details>

**Comment by kunalspathak** [comment]
> related to STR.

I have seen them in the past, but yes, if you can find out what is the cause, that will be good.

**Comment by SwapnilGaikwad** [comment]
All stress tests are passing now.

<details>
<summary> Stress test results </summary>

```
===================Running default===================
------------------- {} -------------------
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.SveDuplicateSelectedScalarToVector_float() : 17
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.SveDuplicateSelectedScalarToVector_double() : 17
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.SveDuplicateSelectedScalarToVector_sbyte() : 17
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.SveDuplicateSelectedScalarToVector_short() : 17
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.SveDuplicateSelectedScalarToVector_int() : 17
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.SveDuplicateSelectedScalarToVector_long() : 17
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.SveDuplicateSelectedScalarToVector_byte() : 17
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.SveDuplicateSelectedScalarToVector_ushort() : 17
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.SveDuplicateSelectedScalarToVector_uint() : 17
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.SveDuplicateSelectedScalarToVector_ulong() : 17
===================Running jitstress===================
------------------- {'JitMinOpts': '1'} -------------------
------------------- {'JitStress': '1'} -------------------
------------------- {'JitStress': '2'} -------------------
------------------- {'JitStress': '1', 'TieredCompilation': '1'} -------------------
------------------- {'JitStress': '2', 'TieredCompilation': '1'} -------------------
------------------- {'TailcallStress': '1'} -------------------
------------------- {'ReadyToRun': '0'} -------------------
===================Running jitstressregs===================
------------------- {'JitStressRegs': '1'} -------------------
------------------- {'JitStressRegs': '2'} -------------------
------------------- {'JitStressRegs': '3'} -------------------
------------------- {'JitStressRegs': '4'} -------------------
------------------- {'JitStressRegs': '8'} -------------------
------------------- {'JitStressRegs': '0x10'} -------------------
------------------- {'JitStressRegs': '0x80'} -------------------
------------------- {'JitStressRegs': '0x1000'} -------------------
------------------- {'JitStressRegs': '0x2000'} -------------------
===================Running jitstress2-jitstressregs===================
------------------- {'JitStress': '2', 'JitStressRegs': '1'} -------------------
------------------- {'JitStress': '2', 'JitStressRegs': '2'} -------------------
------------------- {'JitStress': '2', 'JitStressRegs': '3'} -------------------
------------------- {'JitStress': '2', 'JitStressRegs': '4'} -------------------
------------------- {'JitStress': '2', 'JitStressRegs': '8'} -------------------
------------------- {'JitStress': '2', 'JitStressRegs': '0x10'} -------------------
------------------- {'JitStress': '2', 'JitStressRegs': '0x80'} -------------------
------------------- {'JitStress': '2', 'JitStressRegs': '0x1000'} -------------------
------------------- {'JitStress': '2', 'JitStressRegs': '0x2000'} -------------------
```
</details>

**Review by kunalspathak — CHANGES_REQUESTED** [review]
Added few things to check.

File: src/tests/JIT/HardwareIntrinsics/Arm/Shared/_SveImmUnaryOpTestTemplate.template [hunk 1]
```diff
@@ -0,0 +1,396 @@
...
+            var result = {Isa}.{Method}(
+                Unsafe.Read<{Op1VectorType}<{Op1BaseType}>>(_dataTable.inArrayPtr),
+                {Imm}
+            );
+
+            Unsafe.Write(_dataTable.outArrayPtr, result);
+            ValidateResult(_dataTable.inArrayPtr, {Imm}, _dataTable.outArrayPtr);
+        }
+
+        public void RunBasicScenario_UnsafeRead_InvalidImm()
```
[the remark this answers is not in the document]
Comment (SwapnilGaikwad) [reply src/tests/JIT/HardwareIntrinsics/Arm/Shared/_SveImmUnaryOpTestTemplate.template:191]: Invalid index refers to a value that is permitted but out of bounds on the current system. E.g., bytes permitted index values are between 0 to 63 but on a 256bit SVE length machine, valid indices can be between 256 / 8 - 1 = 0 to 31. Thus, the indices between 32 to 63 would set the destination value to zero.

File: src/coreclr/jit/hwintrinsiccodegenarm64.cpp [hunk 2]
```diff
@@ -784,6 +784,17 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
                     unreached();
             }
         }
+        else if (intrin.id == NI_Sve_DuplicateSelectedScalarToVector)
+        {
```
[the remark this answers is not in the document]
Comment (SwapnilGaikwad) [reply src/coreclr/jit/hwintrinsiccodegenarm64.cpp:788]: Sure, I had tried that but requires more changes and custom handling. I'll update the PR so you can decide whether that suits better.

File: src/coreclr/jit/hwintrinsicarm64.cpp [hunk 3]
```diff
@@ -421,6 +421,10 @@ void HWIntrinsicInfo::lookupImmBounds(
                 immUpperBound = (int)SVE_PATTERN_ALL;
                 break;
 
+            case NI_Sve_DuplicateSelectedScalarToVector:
+                immUpperBound = (512 / (genTypeSize(baseType) * BITS_PER_BYTE)) - 1;
```
[the remark this answers is not in the document]
Comment (SwapnilGaikwad) [reply src/coreclr/jit/hwintrinsicarm64.cpp:425]: We may not able able to use the `getSIMDVectorLength()` here as the `imm` for `DUP` seems special [[1](https://docsmirror.github.io/A64/2023-06/dup_z_zi.html)].
>Is the immediate index, in the range 0 to one less than the number of elements in 512 bits, encoded in "imm2:tsz".

 I didn't find a better helper method for this so did it explicitly.

File: src/tests/Common/GenerateHWIntrinsicTests/GenerateHWIntrinsicTests_Arm.cs [hunk 4]
```diff
@@ -3724,6 +3725,17 @@
...
+    ("SveVecImmUnOpTest.template",        new Dictionary<string, string> { ["TestName"] = "SveDuplicateSelectedScalarToVector_byte",                                                               ["Isa"] = "Sve",           ["LoadIsa"] = "Sve",     ["Method"] = "DuplicateSelectedScalarToVector",                                      ["RetVectorType"] = "Vector",    ["RetBaseType"] = "Byte",    ["Op1VectorType"] = "Vector",    ["Op1BaseType"] = "Byte",                                                                                                                             ["LargestVectorSize"] = "64", ["NextValueOp1"] = "TestLibrary.Generator.GetByte()",                                                                                                                ["InvalidImm"] = "63",   ["Imm"] = "0",  ["ValidateIterResult"] = "result[i] != (testValid ? firstOp[Imm] : 0)", ["GetIterResult"] = "firstOp[Imm]"}),
```
[the remark this answers is not in the document]
Comment (SwapnilGaikwad) [reply src/tests/Common/GenerateHWIntrinsicTests/GenerateHWIntrinsicTests_Arm.cs:3735]: In addition to the `imm` being special in this case here, we have specified a `Max` value in `ConstantExpected` attribute in `Sve.cs`. It leads to `System.ArgumentOutOfRangeException: Specified argument was out of the range of valid values` at runtime for our of range indices, such as values > 63 for bytes and > 15 for ints.

File: src/coreclr/jit/hwintrinsicarm64.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (kunalspathak) [reply src/coreclr/jit/hwintrinsicarm64.cpp:425]: I was actually little confused with that description and then this one:

> The immediate element index is in the range of 0 to 63 (bytes), 31 (halfwords), 15 (words), 7 (doublewords) or 3 (quadwords). 

With ^ description, it sounded me like `getSIMDVectorLength()`, no?

File: src/tests/Common/GenerateHWIntrinsicTests/GenerateHWIntrinsicTests_Arm.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (kunalspathak) [reply src/tests/Common/GenerateHWIntrinsicTests/GenerateHWIntrinsicTests_Arm.cs:3735]: Yes and we want to test that code path :)

File: src/coreclr/jit/hwintrinsiclistarm64sve.h [hunk 5]
```diff
@@ -67,6 +67,7 @@ HARDWARE_INTRINSIC(Sve,           CreateWhileLessThanOrEqualMask8Bit,
...
 HARDWARE_INTRINSIC(Sve,           DotProductBySelectedScalar,                                       -1,      4,      true,  {INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_sve_sdot,       INS_sve_udot,       INS_sve_sdot,       INS_sve_udot,       INS_invalid,        INS_invalid},     HW_Category_SIMDByIndexedElement,  HW_Flag_Scalable|HW_Flag_BaseTypeFromFirstArg|HW_Flag_HasImmediateOperand|HW_Flag_HasRMWSemantics|HW_Flag_LowVectorOperation)
+HARDWARE_INTRINSIC(Sve,           DuplicateSelectedScalarToVector,                                  -1,      2,      true,  {INS_sve_dup,        INS_sve_dup,        INS_sve_dup,        INS_sve_dup,        INS_sve_dup,        INS_sve_dup,        INS_sve_dup,        INS_sve_dup,        INS_sve_dup,        INS_sve_dup},     HW_Category_SIMD,                  HW_Flag_Scalable|HW_Flag_HasImmediateOperand)
```
[the remark this answers is not in the document]
Comment (SwapnilGaikwad) [reply src/coreclr/jit/hwintrinsiclistarm64sve.h]: Done

File: src/coreclr/jit/emitarm64.cpp [hunk 6]
```diff
@@ -7902,7 +7902,9 @@ void emitter::emitIns_R_S(instruction ins, emitAttr attr, regNumber reg1, int va
             {
                 useRegForImm      = true;
                 regNumber rsvdReg = codeGen->rsGetRsvdReg();
-                codeGen->instGen_Set_Reg_To_Imm(EA_PTRSIZE, rsvdReg, imm);
+                codeGen->instGen_Set_Reg_To_Base_Plus_Imm(EA_PTRSIZE, rsvdReg, reg2, imm);
+                reg2 = rsvdReg;
```
[the remark this answers is not in the document]
Comment (SwapnilGaikwad) [reply src/coreclr/jit/emitarm64.cpp]: Done

File: src/coreclr/jit/hwintrinsiccodegenarm64.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (SwapnilGaikwad) [reply src/coreclr/jit/hwintrinsiccodegenarm64.cpp]: Done

File: src/tests/Common/GenerateHWIntrinsicTests/GenerateHWIntrinsicTests_Arm.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (SwapnilGaikwad) [reply src/tests/Common/GenerateHWIntrinsicTests/GenerateHWIntrinsicTests_Arm.cs]: Done 👍

File: src/tests/Common/GenerateHWIntrinsicTests/GenerateHWIntrinsicTests_Arm.cs [hunk 7]
```diff
@@ -3724,6 +3725,17 @@
...
+
+    ("SveVecImmUnOpTest.template",        new Dictionary<string, string> { ["TestName"] = "SveDuplicateSelectedScalarToVector_float",                                                              ["Isa"] = "Sve",           ["LoadIsa"] = "Sve",     ["Method"] = "DuplicateSelectedScalarToVector",                                      ["RetVectorType"] = "Vector",    ["RetBaseType"] = "Single",  ["Op1VectorType"] = "Vector",    ["Op1BaseType"] = "Single",                                                                                                                           ["LargestVectorSize"] = "64", ["NextValueOp1"] = "TestLibrary.Generator.GetSingle()",                                                                                                              ["InvalidImm"] = "15",   ["Imm"] = "0",  ["ValidateIterResult"] = "result[i] != (testValid ? firstOp[Imm] : 0)", ["GetIterResult"] = "firstOp[Imm]"}),
```
[the remark this answers is not in the document]
Comment (SwapnilGaikwad) [reply src/tests/Common/GenerateHWIntrinsicTests/GenerateHWIntrinsicTests_Arm.cs]: Done

**Comment by SwapnilGaikwad** [comment]
[JIT dump](https://gist.github.com/SwapnilGaikwad/8e1e84659594683a898b6302ffd4f693#file-sve_dup_reg_allocation_issue-dump) for a failing method.

**Comment by kunalspathak** [comment]
> [JIT dump](https://gist.github.com/SwapnilGaikwad/8e1e84659594683a898b6302ffd4f693#file-sve_dup_reg_allocation_issue-dump) for a failing method.

@SwapnilGaikwad - Are you sure this one is the right dump? I see the compilation succeed in this dump.

**Comment by kunalspathak** [comment]
> Do you have any suggestions on how to fix them?

This is to do with that a node is of type `TYP_FLOAT` and we assigned it candidates of `TYP_MASK` or vice-versa. It might not be directly related to your PR, but something that your PR exposed specially after the LSRA work of predicate register is complete. If you share the correct dump, I can quickly tell for which node it is getting wrong, and you can then track in `lsraarm64.cpp` why that is happening.

**Comment by SwapnilGaikwad** [comment]
Hi Kunal, apologies. It was dump of the parent test method that emitted a call to `Sve.DuplicateSelectedScalarToVector`.
I have updated[ the gist](https://gist.github.com/SwapnilGaikwad/8e1e84659594683a898b6302ffd4f693) with the correct dump now.

**Comment by kunalspathak** [comment]
> Hi Kunal, apologies. It was dump of the parent test method that emitted a call to `Sve.DuplicateSelectedScalarToVector`. I have updated[ the gist](https://gist.github.com/SwapnilGaikwad/8e1e84659594683a898b6302ffd4f693) with the correct dump now.

```
DefList: { N004.t0. LCL_VAR; N014.t1. LCL_VAR }
N016 (  4,  4) [000009] -----------                         *  CAST      int <- ubyte <- int REG NA
<RefPosition #7   @16  RefTypeUse <Ivl:3> BB01 regmask=[x0-xip0 x19-x28] minReg=1 last wt=100.00>
Interval  4: int RefPositions {} physReg:NA Preferences=[x0-xip0 x19-x28] Aversions=[]
<RefPosition #8   @17  RefTypeDef <Ivl:4> CAST BB01 regmask=[x0-xip0 x19-x28] minReg=1 wt=400.00>

DefList: { N004.t0. LCL_VAR; N016.t9. CAST }
N018 ( 17, 20) [000006] ---X-------                         *  HWINTRINSIC simd16 short DuplicateSelectedScalarToVector REG NA
Interval  5: int RefPositions {} physReg:NA Preferences=[x0-xip0 x19-x28] Aversions=[]
<RefPosition #9   @18  RefTypeDef <Ivl:5 internal> HWINTRINSIC BB01 regmask=[x0-xip0 x19-x28] minReg=1 wt=400.00>
<RefPosition #10  @18  RefTypeUse <Ivl:0> BB01 regmask=[allFloat] minReg=1 last wt=100.00>
<RefPosition #11  @18  RefTypeUse <Ivl:4> BB01 regmask=[d0-d15] minReg=1 last wt=100.00>
<RefPosition #12  @18  RefTypeUse <Ivl:5 internal> HWINTRINSIC BB01 regmask=[x0-xip0 x19-x28] minReg=1 last wt=400.00>
Interval  6: simd16 RefPositions {} physReg:NA Preferences=[allFloat] Aversions=[]
<RefPosition #13  @19  RefTypeDef <Ivl:6> HWINTRINSIC BB01 regmask=[allFloat] minReg=1 wt=400.00>
```

I think the problem is in `<Ivl:4>`. For def `RefPosition #8`, the mask created is GPR, but the use `RefPosition #11`, the mask if `d0-d15` which is GPR, which seems wrong. So, you might want to check by putting breakpoint in `newRefPositionRaw()` for `newRP->rpNum == 11` why the mask is not gpr.

**Comment by kunalspathak** [comment]
@SwapnilGaikwad  - I assume you don't need anything from me for this PR?

**Comment by SwapnilGaikwad** [comment]
> why the mask is not gpr

Hi @kunalspathak , your suggestion was very useful to identify the issue. The mask was hardcoded to simd registers for op2.

<details>
<summary> All stress tests pass </summary>

```
```
</details>

File: src/coreclr/jit/lsraarm64.cpp [hunk 8]
```diff
@@ -1678,7 +1682,14 @@ int LinearScan::BuildHWIntrinsic(GenTreeHWIntrinsic* intrinsicTree, int* pDstCou
         {
             assert(!isRMW);
 
-            srcCount += BuildOperandUses(intrin.op2, RBM_ASIMD_INDEXED_H_ELEMENT_ALLOWED_REGS.GetFloatRegSet());
+            if (intrin.id == NI_Sve_DuplicateSelectedScalarToVector)
```
Comment (kunalspathak) [inline src/coreclr/jit/lsraarm64.cpp:1685]: hhm, in both cases, we were using float register candidates and with your fix, we are just using reduced float register candidates. how was that causing an issue where we were just associating wrong type of candidates to a node?

Comment (kunalspathak) [reply src/coreclr/jit/lsraarm64.cpp:1685]: Or was it the case, where type of candidates were correct, but the actual set should have been different, which you are handling it here?

**Review by kunalspathak — COMMENTED** [review]
Looks good overall. Have a question about the change in `codegenarm64.cpp`.

Also, how are we exposing [DUP (immediate) ](https://docsmirror.github.io/A64/2023-06/dup_z_i.html) and [DUP (scalar) ](https://docsmirror.github.io/A64/2023-06/dup_z_r.html)?

File: src/coreclr/jit/emitarm64.cpp [hunk 9]
```diff
@@ -8150,9 +8153,11 @@ void emitter::emitIns_S_R(instruction ins, emitAttr attr, regNumber reg1, int va
             {
                 useRegForImm      = true;
                 regNumber rsvdReg = codeGen->rsGetRsvdReg();
-                codeGen->instGen_Set_Reg_To_Imm(EA_PTRSIZE, rsvdReg, imm);
+                // For larger imm values (> 9 bits), calculate base + imm in a reserved register first.
+                codeGen->instGen_Set_Reg_To_Base_Plus_Imm(EA_PTRSIZE, rsvdReg, reg2, imm);
```
[the remark this answers is not in the document]
Comment (kunalspathak) [reply src/coreclr/jit/emitarm64.cpp:8157]: I might actually need it for https://github.com/dotnet/runtime/pull/104065/files#diff-2b2c8b9011607926410624d6f81613fad7b74c6e0516d578675a8b792998fe4fR7893-R7896, but I am curious if you found a repro, for which you had to add change here.

[the remark this answers is not in the document]
Comment (SwapnilGaikwad) [reply src/coreclr/jit/emitarm64.cpp:8157]: This is not related to DUP directly. However, tests trigger a scenario where we end up in having a store with a larger immediate.

**Comment by kunalspathak** [comment]
> Also, how are we exposing [DUP (immediate) ](https://docsmirror.github.io/A64/2023-06/dup_z_i.html)and [DUP (scalar) ](https://docsmirror.github.io/A64/2023-06/dup_z_r.html)?

Spoke offline that we will use these in future when we intrinsify `Vector.Create` or similar scenario for sve.

**Review by kunalspathak — APPROVED** [review]
LGTM

**Pull request merged**